### PR TITLE
feat: Add PUT APIs for the userdetails feature in the dashboard recipe

### PR DESCRIPTION
**Pull request by nkshah2**
## Summary of change

Adds the following APIs:
- `/user DELETE`: Delete a user
- `/user/email/verify PUT`: Set the email verification status of a user
- `/user/metadata PUT`: Set the metadata for a user
- `/user/password PUT`: Change the users password
- `/user PUT`: Update user information

## Related issues

- https://github.com/supertokens/supertokens-node/pull/432
- Full API spec: https://github.com/supertokens/dashboard/blob/master/api_spec.yaml (Copy the raw file and paste it into https://editor.swagger.io/ to view it)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens/constants.go`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `supertokens/constants.go > version variable`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: recipe/dashboard/api/userdetails/userPut.go [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+	Status string
+	Error string
+}
+
+type updatePhoneResponse struct {
+	Status string
+	Error string
+}
+
+type userPutRequestBody struct {
```
Comment (sattvikc) [inline recipe/dashboard/api/userdetails/userPut.go:49]: JSON fields for unmarshall

File: recipe/dashboard/api/userdetails/userEmailVerifyPut.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+)
+
+type userEmailVerifyPutResponse struct {
+	Status string `json:"status"`
+}
+
+type userEmailVerifyPutRequestBody struct {
+	UserID string
+	Verified bool
+}
```
Comment (sattvikc) [inline recipe/dashboard/api/userdetails/userEmailVerifyPut.go:34]: JSON fields for unmarshall

File: recipe/dashboard/api/userdetails/userMetadataPut.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+)
+
+type userMetadataPutResponse struct {
+	Status string `json:"string"`
+}
+
+type userMetaDataRequestBody struct {
+	UserId string
+	Data string
+}
```
Comment (sattvikc) [inline recipe/dashboard/api/userdetails/userMetadataPut.go:33]: JSON fields for unmarshall

File: recipe/dashboard/api/userdetails/userPasswordPut.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+type userPasswordPutResponse struct {
+	Status string `json:"status,omitempty"`
+	Error string `json:"error,omitempty"`
+}
+
+type userPasswordPutRequestBody struct {
+	UserId string
+	NewPassword string
+}
```
Comment (sattvikc) [inline recipe/dashboard/api/userdetails/userPasswordPut.go:37]: JSON fields for unmarshall

**Review by sattvikc — CHANGES_REQUESTED** [review]

File: recipe/dashboard/api/userdetails/userPut.go [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+
+type userPutResponse struct {
+	Status string `json:"status"`
+	Error string `json:"error,omitempty"`
+}
+
+type updateEmailResponse struct {
+	Status string
+	Error string
+}
```
Comment (sattvikc) [inline recipe/dashboard/api/userdetails/userPut.go:42]: missing json

File: recipe/dashboard/api/userdetails/userPut.go [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+
+type updatePhoneResponse struct {
+	Status string
+	Error string
+}
```
Comment (sattvikc) [inline recipe/dashboard/api/userdetails/userPut.go:47]: Missing json

**Review by sattvikc — CHANGES_REQUESTED** [review]

File: recipe/dashboard/api/userdetails/userPut.go (hunk 5 above)
Comment (nkshah2) [reply recipe/dashboard/api/userdetails/userPut.go]: This is never sent to the frontend

File: recipe/dashboard/api/userdetails/userPut.go (hunk 6 above)
Comment (nkshah2) [reply recipe/dashboard/api/userdetails/userPut.go]: This is never sent to the frontend

**Review by sattvikc — APPROVED** [review]

**Pull request merged**